File: frpc/csharp-peer/Models/Stream.cs
using Fomoxa;

namespace Fomoxa.Rpc.Demo.Models
{
    [Network]
    [Codec("rpc")]
    public class UploadOpen
    {
        [Network("string")]
        [Codec("rpc")]
        public string Name { get; set; } = "";
    }

    [Network]
    [Codec("rpc")]
    public class UploadChunk
    {
        [Network("u32")]
        [Codec("rpc")]
        public uint Size { get; set; }
    }

    [Network]
    [Codec("rpc")]
    public class UploadDone
    {
        [Network("string")]
        [Codec("rpc")]
        public string Summary { get; set; } = "";
    }

    [Network]
    [Codec("rpc")]
    public class ChatOpen
    {
        [Network("string")]
        [Codec("rpc")]
        public string Room { get; set; } = "";
    }

    [Network]
    [Codec("rpc")]
    public class ChatSaid
    {
        [Network("string")]
        [Codec("rpc")]
        public string Line { get; set; } = "";
    }

    [Network]
    [Codec("rpc")]
    public class ChatHeard
    {
        [Network("string")]
        [Codec("rpc")]
        public string Line { get; set; } = "";
    }

    [Network]
    [Codec("rpc")]
    public class TicksRequest
    {
    }

    [Network]
    [Codec("rpc")]
    public class TicksItem
    {
        [Network("u32")]
        [Codec("rpc")]
        public uint Value { get; set; }
    }
}

File: frpc/rust-peer/src/demo/src/models/stream.rs
#[allow(unused_imports)]
use fomoxa_attributes::{codec, network};

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct UploadOpen {
    #[network(string)]
    #[codec(rpc)]
    pub name: String,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct UploadChunk {
    #[network(u32)]
    #[codec(rpc)]
    pub size: u32,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct UploadDone {
    #[network(string)]
    #[codec(rpc)]
    pub summary: String,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct ChatOpen {
    #[network(string)]
    #[codec(rpc)]
    pub room: String,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct ChatSaid {
    #[network(string)]
    #[codec(rpc)]
    pub line: String,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct ChatHeard {
    #[network(string)]
    #[codec(rpc)]
    pub line: String,
}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct TicksRequest {}

#[derive(Debug, Default, Clone)]
#[network]
#[codec(rpc)]
pub struct TicksItem {
    #[network(u32)]
    #[codec(rpc)]
    pub value: u32,
}
